src/main/java/com/sourcegraph/webhook/WebhookRouter.java

```diff
@@ -0,0 +1,46 @@
+package com.sourcegraph.webhook;
+
+import com.google.gson.Gson;
+import com.sourcegraph.webhook.registry.Webhook;
+import com.sourcegraph.webhook.registry.WebhookRegistry;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+@Path("/webhook")
+public class WebhookRouter {
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response get() {
+        List<Webhook> hooks = WebhookRegistry.getWebhooks();
+        String resp = new Gson().toJson(hooks);
+        return Response.ok(resp).build();
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public Response put(
+            @FormParam("scope") String scope,
+            @FormParam("identifier") String identifier,
+            @FormParam("events") String events,
+            @FormParam("external") String external) {
```

Comment:
We need a `secret` as well, so that events are sent with a HMAC header signature of the payload for authentication purposes. Otherwise we have no way to authenticate incoming webhook requests in Sourcegraph. See GitHub's implementation for reference: https://github.com/sourcegraph/sourcegraph/blob/master/enterprise/pkg/a8n/webhooks_test.go#L244